.github/workflows/nonregression.yml

```diff
@@ -35,16 +158,36 @@ jobs:
           docker pull ghcr.io/newrelic/deployer:latest
           docker images ghcr.io/newrelic/deployer:latest
 
-      - name: Run deployer and cleanup
+      - name: Run deployer
+        id: deployerRun
         run: |
+          set -e
+          testDefinitionFile=$(echo $MATRIX | jq -c -r '.testDefinitionFile')
+          echo $testDefinitionFile
           docker run -i\
             -v ${{ github.workspace }}/configs/:/mnt/deployer/configs/\
             -v ${{ github.workspace }}/test/:/mnt/deployer/test/\
-            --entrypoint ruby ghcr.io/newrelic/deployer:latest batch.rb -c configs/gitusdkr${{ github.run_id }}.json -d test/definitions/ -s 20 -m deploy -l debug
+            --entrypoint ruby ghcr.io/newrelic/deployer:latest main.rb -c configs/gitusdkr${{ github.run_id }}.json -d $testDefinitionFile -l debug
+          echo ::set-output name=exit_status::$?
+      - name: Teardown any previous deployment
```

Comment:
It would be great to see how we could offload the teardown to another GH run, for example this is the job that clean the resources after running the regression tests https://github.com/newrelic/open-install-library/blob/main/.github/workflows/cleanup.yml
This would speed up the tests further. This can be done outside the scope of this PR though.

Reply:
Agreed! I would be happy to take a look at using the cleanup/-eu workflows rather than triggering teardown in each test individually in another PR.